### PR TITLE
auditctl: support reporting backlog_wait_time_actual

### PR DESCRIPTION
**Pull request by maxenglander**
The kernel audit subsystem can report the sum of time it has waited
while the backlog limit is exceeded. This commits adds support for
reporting that sum as `backlog_wait_time_actual` with `auditctl -s`,
and resetting that value with
`auditctl --reset_backlog_wait_time_actual`.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -88,6 +88,7 @@
 - Speedup ausearch/report parsing RAW logging format by caching uid/name lookup
 - Change auparse python bindings to shared object (Issue #121)
 - Add error messages for watch permissions
+- Report backlog_wait_sum with auditctl -s
```
Comment (maxenglander) [inline ChangeLog:91]:
```suggestion
- Report and reset actual backlog wait time
```

**Comment by maxenglander** [comment]
Hi @stevegrubb looks like Paul is OK with the kernel patch. Does this user-space patch look OK to you?

**Comment by maxenglander** [comment]
Hi @stevegrubb now that the `backlog_wait_time_total` field has been merged into https://github.com/linux-audit/audit-kernel/commit/b43870c74f3fdf0cd06bf5f1b7a5ed70a2cd4ed2 and made its way upstream to Torvald's tree, does it make sense to get this merged into the userspace?

**Comment by stevegrubb** [comment]
I was just looking at my kernel's audit,h file and I do not see AUDIT_STATUS_BACKLOG_WAIT_TIME_ACTUAL. Wonder where it went?

**Comment by stevegrubb** [comment]
Actually, just found it in current audit tree.

**Comment by stevegrubb** [comment]
Hmm. When I run the command, I see backlog_wait_time_actual: 0. 
However, auditctl -s does not show backlog_wait_time_actual.

**Comment by maxenglander** [comment]
Hey @stevegrubb wanted to follow up here to let you know I pushed a commit based on the discussion in the mailing list. I'm not 100% sure I'm following what you and Richard had in mind, but here's what I tested to verify:

 * Install the stable-5.9 audit kernel, the most recent commit (linux-audit/audit-kernel@b43870c) on that branch has the backlog_wait_time_actual changes
 * Install the userspace in this branch (latest commit 62de539)
 * Verify that backlog_wait_time_actual shows up when I do auditctl -s
 * Restart and select 4.15 kernel with grub
 * Verify that backlog_wait_time_actual does not show up when I do auditctl -s

> Hmm. When I run the command, I see backlog_wait_time_actual: 0.
However, auditctl -s does not show backlog_wait_time_actual.

Hm, strange. What operating system and hardware do you have? What commits for kernel and userspace? I can try to reproduce with some info.

**Comment by stevegrubb** [comment]
Generally, including ChangeLog will cause problems merging. I always add an entry by hand.

**Comment by stevegrubb** [comment]
Hunk 3 of auditctl.c doesn't apply cleanly. Another patch changed that area first. I can apply this patch manually (I have a fixed copy). Or if you'd like, you can fix the conflicts and force-push the changed patch. This patch does appear to work cleanly on 5.8 and 5.9 kernels.

**Comment by stevegrubb** [comment]
This patch was committed from the command line. Closing the pull request. Thanks for the patch!

**Comment by maxenglander** [comment]
Ah my mistake. Thanks for sorting it out and getting it merge!

**Pull request closed without merging**